### PR TITLE
Fix compatibility with pre-4.15 RPM versions with DynamicBuildRequires

### PR DESCRIPTION
**Pull request by ignatenkobrain**
Signed-off-by: Igor Gnatenko <i.gnatenko.brain@gmail.com>

### CODE REVIEW

**Comment by ignatenkobrain** [comment]
@xsuchy can you release last few commits in a new version or backport in Fedora please?

File: mock/py/mockbuild/backend.py [hunk 1]
```diff
@@ -283,7 +283,8 @@ def build(self, srpm, timeout, check=True, spec=None):
 
             # Check if we will have dynamic BuildRequires, but do not allow it
             hdr = next(util.yieldSrpmHeaders((rebuilt_srpm,)))
-            dynamic_buildreqs = 'rpmlib(DynamicBuildRequires)' in hdr[rpm.RPMTAG_REQUIRES]
+            requires = {util._to_text(req) for req in hdr[rpm.RPMTAG_REQUIRES]}
```
Comment (xsuchy) [inline mock/py/mockbuild/backend.py]: Why { (dictionary) ? Shouldn't it be list or tuple?

Comment (ignatenkobrain) [reply mock/py/mockbuild/backend.py]: It is a set, not a dictionary

Comment (ignatenkobrain) [reply mock/py/mockbuild/backend.py]: And checking if element is in set is faster than in list...

Comment (xsuchy) [reply mock/py/mockbuild/backend.py]: ah, ok - this was actually new for me.

**Comment by ignatenkobrain** [comment]
@xsuchy :) Could you please release new mock so that we can get it in Fedora infra instead of backporting separate patches?

**Pull request merged**